### DISCUSSION TITLE
Scan and use safe strings from google closure compiler output as sources

### DISCUSSION THREAD

**Discussion by agorgl** [tailwindlabs/tailwindcss#9780] [Ideas]
Currently if tailwindcss jit scans through optimized sources generated by google closure compiler it mostly works.
There is a case when some tailwindcss classes contain characters that are replaced by closure compiler, due to it's 'trustedStrings' functionality.

A good example is data variants, where for example `data-[edit=false]` may become `data-[edit\x3dfalse]`
This happens here in the google closure compiler:
https://github.com/google/closure-compiler/blob/060e4905d3182fbaaff93449e4cf4ed787314355/src/com/google/javascript/jscomp/CodeGenerator.java#L2156

This is also the case for all clojurescript based projects as the release builds pass through the google closure compiler.
Ideally the tailwindcss jit should scan for 'safe/trusted' variants as generated by the google closure compiler too.

Relevant: https://github.com/thheller/shadow-cljs/commit/5cf0ea00a08e4de75d30465a9d84b8ad6846e2f8

**Comment by adamwathan** [comment]
In the example you mentioned does the final HTML contain `data-[edit=false]` when inspected in the browser, or `data-[edit\x3dfalse]`? Can you create a very small demo project that shows some of these situations?

**Reply by agorgl** [reply]
Well the final html is actually produced in the runtime (its a SPA app) and yes, it contains `data-[edit=false]` as it should.
The problem is that the `.js` that contains the javascript code that creates the rendered html page contains it escaped as `data-[edit\x3dfalse]`
